config/routes.rb

```diff
@@ -296,7 +296,11 @@
         :via => :get
   ####
 
+  #### PublicBodyChangeRequest controller
   resource :change_request, :only => [:new, :create], :controller => 'public_body_change_requests'
+  match 'change_request/new/:body' => 'public_body_change_requests#new',
+        :as => :new_change_request_body,
+        :via => :get
```

Comment:
Rather than hacking in a `match` route, I think it would be better to use the Rails conventions and create a nested resource:

```diff
diff --git a/config/routes.rb b/config/routes.rb
index 5591811..a86a2dc 100644
--- a/config/routes.rb
+++ b/config/routes.rb
@@ -297,10 +297,11 @@ Alaveteli::Application.routes.draw do
   ####

   #### PublicBodyChangeRequest controller
-  resource :change_request, :only => [:new, :create], :controller => 'public_body_change_requests'
-  match 'change_request/new/:body' => 'public_body_change_requests#new',
-        :as => :new_change_request_body,
-        :via => :get
+  resources :public_bodies, :only => [] do
+    resources :change_requests,
+              :only => [:new, :create],
+              :controller => 'public_body_change_requests'
+  end

   #### Comment controller
   match '/annotate/request/:url_title' => 'comment#new',
```

```sh
$ be rake routes | g change_request
     public_body_change_requests POST           /public_bodies/:public_body_id/change_requests(.:format)                public_body_change_requests#create
  new_public_body_change_request GET            /public_bodies/:public_body_id/change_requests/new(.:format)            public_body_change_requests#new
```

This has a nice minor benefit of being able to drop the custom url param in the form:

```diff
diff --git a/app/views/public_body_change_requests/new.html.erb b/app/views/public_body_change_requests/new.html.erb
index 12c5879..5a89396 100644
--- a/app/views/public_body_change_requests/new.html.erb
+++ b/app/views/public_body_change_requests/new.html.erb
@@ -2,7 +2,7 @@

 <%= foi_error_messages_for :change_request %>

-<%= form_for(@change_request, :url => change_request_path)  do |f| %>
+<%= form_for(@change_request) do |f| %>
   <% if not @user %>
     <p>
       <label class="form_label" for="user_name">
```